python/astro_metadata_translator/translator.py

```diff
@@ -807,6 +807,44 @@ def to_observation_reason(self):
             return "science"
         return "unknown"
 
+    @cache_translation
+    def to_observing_day(self):
+        """Return the YYYYMMDD integer corresponding to the observing day.
+
+        Base class implementation uses the TAI date of the start of the
+        observation.
+
+        Returns
+        -------
+        day : `int`
+            The observing day as an integer of form YYYYMMDD. If the header
+            is broken and is unable to obtain a date of observation, ``0``
+            is returned and the assumption is made that the problem will
+            be caught elsewhere.
+        """
+        datetime_begin = self.to_datetime_begin()
+        if datetime_begin is None:
```

Comment:
Why isn't this a failure?  Or is that handled elsewhere?

Reply:
datetime begin returning None is not technically a failure (as opposed to the method crashing). It might be undesirable but I'd rather not have the base implementation of this translation method fall over without warning. An instrument can override it if they are worried and if the caller is worried about a None for date time begin they will be complaining about that regardless of what happens here.